src/main/java/com/flansmod/client/model/RenderGun.java

```diff
@@ -556,332 +551,325 @@ else if (!empty && !animations.reloading)
-			{
-				GL11.glPushMatrix();
-				ModelFlash flash = type.flashModel;
-				GL11.glScalef(model.flashScale, model.flashScale, model.flashScale);
-				{
-					Vector3f base = model.muzzleFlashPoint == null ? Vector3f.Zero : model.muzzleFlashPoint;
-					if (barrelAttachment != null) {
-						Vector3f barrelOffset = (barrelAttachment.model != null && barrelAttachment.model.attachmentFlashOffset != null) ? barrelAttachment.model.attachmentFlashOffset : Vector3f.Zero;
-						GL11.glTranslatef(base.x + barrelOffset.x,
-											base.y + barrelOffset.y,
-											base.z + barrelOffset.z);
-					} else {
-						Vector3f defaultOffset = model.defaultBarrelFlashPoint == null ? Vector3f.Zero : model.defaultBarrelFlashPoint;
-
-						GL11.glTranslatef(base.x + defaultOffset.x,
-								base.y + defaultOffset.y,
-								base.z + defaultOffset.z);
-					}
-
-					renderEngine.bindTexture(FlansModResourceHandler.getAuxiliaryTexture(type.flashTexture));
-					ModelGun.glowOn();
-					flash.renderFlash(f, animations.flashInt);
-					ModelGun.glowOff();
-					renderEngine.bindTexture(FlansModResourceHandler.getPaintjobTexture(type.getPaintjob(item.getItemDamage())));
-				}
-				GL11.glPopMatrix();
-			}
-
-
-			// Render various shoot / reload animated parts
+			//Handle muzzle flash stuffs
+			renderMuzzleFlash(animations, model, type, barrelAttachment, item, f);
+			
 			// Render the slide and charge action
 			if (slideAttachment == null)
-			{
-				GL11.glPushMatrix();
-				{
-					if (!type.getSecondaryFire(item)) {
-						GL11.glTranslatef(
-								-(animations.lastGunSlide + (animations.gunSlide - animations.lastGunSlide) * smoothing)
-										* model.gunSlideDistance,
-								0F, 0F);
-						GL11.glTranslatef(-(1 - Math.abs(
-								animations.lastCharged + (animations.charged - animations.lastCharged) * smoothing))
-								* model.chargeHandleDistance, 0F, 0F);
-
-					}
-					model.renderSlide(f);
-					if (scopeAttachment == null && model.scopeIsOnSlide)
-						model.renderDefaultScope(f);
-				}
-				GL11.glPopMatrix();
-			}
+				renderSlideModels(scopeAttachment, type, animations, model, f, item);
+
+			//Render break action(s)
+			renderBreakAction(model, scopeAttachment, reloadRotate, f);
+			//Render hammer action(s)
+			renderHammer(model, animations, f);
+			//Render pump action(s)
+			renderPumpAction(model, animations, pumpAttachment, f, gripAttachment, type, gadgetAttachment);
+			//Render charge handle(s)
+			renderChargeHandle(model, animations, f);
+			//Render minigun barrels
+			renderMinigunBarrels(type, model, animations, f);
+			// Render the revolver barrel
+			renderRevolverBarrel(model, reloadRotate, f);
+			//Render gun ammo
+			renderAmmo(model, type, gripAttachment, item, empty, animations, reloadRotate, rtype, f, gripItemStack);
+			// Render a static model of the ammo NOT being reloaded
+			//this seems to not be required at all
+			//renderStaticAmmo(type, item, model, f, gripAttachment, gripItemStack);
```

Comment:
I couldn't figure out what this bit was actually doing, it doesn't seem to change anything if it's commented out. Thoughts?

Reply:
Looks like it loads ammo models for underbarrels, I didn't know underbarrels could have ammo models!?

Reply:
Ah okay yeah you're probably right

Reply:
Alr